libraries/node-core-library/src/test/Executable.test.ts

```diff
@@ -298,17 +284,44 @@ describe('Executable process tests', () => {
       currentWorkingDirectory: executableFolder
     });
     const result: IWaitForExitResult<string> = await Executable.waitForExitAsync(childProcess, {
-      encoding: 'utf8',
-      throwOnNonZeroExitCode: false
+      encoding: 'utf8'
     });
     expect(result.exitCode).toEqual(1);
+    expect(result.signal).toBeNull();
     expect(typeof result.stdout).toEqual('string');
     expect(typeof result.stderr).toEqual('string');
-    expect(result.stdout.indexOf('Executing javascript-file.js with args:')).toEqual(-1);
+    expect(result.stdout).toMatch(/^Executing npm-binary-wrapper(\.cmd)? with args:/);
     expect(result.stderr.endsWith('This is a failure'));
   });
 
-  test('Executable.runToCompletion(Executable.spawn("npm-binary-wrapper")) failure with throw', async () => {
+  test('Executable.runToCompletion(Executable.spawn("no-terminate")) killed', async () => {
+    const executablePath: string = path.join(executableFolder, 'no-terminate', 'javascript-file.js');
+    const childProcess: child_process.ChildProcess = Executable.spawn(
+      process.argv0,
+      [executablePath, '1', '2', '3'],
+      {
+        environment,
+        currentWorkingDirectory: executableFolder
+      }
+    );
+    const startTime: number = Date.now();
+    await Async.sleepAsync(50); // Wait for the process to boot and print
+    const killResult: boolean = childProcess.kill('SIGTERM');
+    const result: IWaitForExitResult<string> = await Executable.waitForExitAsync(childProcess, {
+      encoding: 'utf8'
+    });
+    const endTime: number = Date.now();
+    expect(killResult).toBe(true);
+    expect(result.signal).toBe('SIGTERM');
+    expect(result.exitCode).toBeNull();
+    expect(typeof result.stdout).toEqual('string');
+    expect(typeof result.stderr).toEqual('string');
+    expect(result.stdout).toMatch(/^Executing no-terminate with args:/);
+    expect(result.stderr.includes('This process never terminates')).toBe(true);
+    expect(endTime - startTime).toBeLessThan(500);
```

Comment:
This feels like something that could possibly become flaky on different hardware

Reply:
If it becomes a problem, we'll change this.